File: 2 C#-Grundlagen und GUIs/8 Digitaluhr_V2/Digitaluhr_V2/Uhrenanzeige.cs
namespace Digitaluhr_V2
{
    public class Uhrenanzeige
    {
        private readonly Nummernanzeige minuten;
        private readonly Nummernanzeige sekunden;
        private readonly Nummernanzeige stunden;

        public Uhrenanzeige()
        {
            stunden = new Nummernanzeige(23);
            minuten = new Nummernanzeige(59);
            sekunden = new Nummernanzeige(59);
        }

        public override string ToString()
        {
            return stunden + " : " + minuten + " : " + sekunden;
        }

        public void SetzeZeit(int std, int min, int sek)
        {
            stunden.Wert = std;
            minuten.Wert = min;
            sekunden.Wert = sek;
        }

        public void ErhoeheZeit()
        {
            sekunden.ErhoeheWert();
            if (sekunden.Wert == 0)
            {
                minuten.ErhoeheWert();
                if (minuten.Wert == 0) stunden.ErhoeheWert();
            }
        }
    }
}

File: Informatik/5 Testen und Debugging/(1) Digitaluhr_V1/Uhrenanzeige.java
/**
 * Die Klasse Uhrenanzeige simuliert eine Digitaluhr,
 * ... zur Darstellung von Stunden, Minuten und Sekunden. 
 * Autor:   Martin Lentz
 * Version: 01.10.2014
 */
public class Uhrenanzeige
{
    private Nummernanzeige stunden, minuten, sekunden;
    
    public Uhrenanzeige()
    {
        stunden = new Nummernanzeige(23);
        minuten = new Nummernanzeige(59);
        sekunden = new Nummernanzeige(59);
    }

    // Liefert die aktuelle Zeit 
    public String gibZeit()
    {
        return stunden.gibWert()+" : "+minuten.gibWert()+" : "+sekunden.gibWert();
    }
    
    // Stellt neue Zeit ein
    public void setzeZeit(int stunden, int minuten, int sekunden)
    {
        this.stunden.setzeWert(stunden);
        this.minuten.setzeWert(minuten);
        this.sekunden.setzeWert(sekunden);
    }
    
    // Erhöht die Zeit um einen Takt, d. h. um eine Sekunde 
    public void erhoeheZeit()
    {
        sekunden.erhoeheWert();
        if (sekunden.gibWert() == 0)
        {
            minuten.erhoeheWert();
            if (minuten.gibWert() == 0)
            {
                stunden.erhoeheWert();
            }
        }
    }
}